src/libcore/macros.rs

```diff
@@ -56,7 +64,8 @@ macro_rules! panic {
 macro_rules! assert {
     ($cond:expr) => (
         if !$cond {
-            panic!(concat!("assertion failed: ", stringify!($cond)))
```

Comment:
It looks like there's some logic to handle this below, so just to confirm is this transformation necessary?